src/sif/truckcost.cc

```diff
@@ -484,7 +484,9 @@ Cost TruckCost::EdgeCost(const baldr::DirectedEdge* edge,
                                          &flow_sources, time_info.seconds_from_now)
                         : fixed_speed_;
 
-  auto final_speed = std::min(edge_speed, top_speed_);
+  auto final_speed = std::min(std::min(edge_speed, edge->truck_speed() ? edge->truck_speed()
```

Comment:
Right now we're getting the `DirectedEdge`'s  `truck_speed()` inside `GetSpeed`. I'll try to pull that out of there so that we only call it here in `EdgeCost`.